tests/conftest.py

```diff
@@ -231,6 +232,62 @@ def _benchmark_time():
     return _benchmark_time()
 
 
+@pytest.fixture(scope="function")
+def benchmark_coiled_prometheus(test_run_benchmark):
+    """Benchmark the coiled prometheus metrics of executing some code.
+
+    Yields
+    ------
+    Context manager that records the coiled prometheus metrics of executing
+    the ``with`` statement if run as part of a benchmark, or does nothing otherwise.
+
+    Example
+    -------
+    .. code-block:: python
+
+        def test_something(benchmark_coiled_prometheus):
+            with benchmark_coiled_prometheus:
+                do_something()
+    """
+
+    @contextlib.contextmanager
+    def _benchmark_coiled_prometheus(client):
+        if not test_run_benchmark:
+            yield
+        else:
+            start = math.floor(time.time())
+            yield
+            end = math.ceil(time.time())
+            cluster = client.cluster
+            test_run_benchmark.scheduler_memory_max = cluster.get_aggregated_metric(
+                query="(host_memory_total-host_memory_available)&scheduler",
+                over_time="max",
+                start_ts=start,
+                end_ts=end,
+            )
+            test_run_benchmark.scheduler_cpu_avg = cluster.get_aggregated_metric(
+                query="cpu_rate&scheduler",
+                over_time="avg",
+                start_ts=start,
+                end_ts=end,
+            )
+            # TODO: Do we want to have plain max or 80th percentile?
+            test_run_benchmark.worker_max_tick = cluster.get_aggregated_metric(
+                query="worker_max_tick|pct80",
```

Comment:
just to be clear, `pct80` here means that at each time you ignore the worst 20% of workers.

aggregations in `query` are applied at each time, aggregations in `over_time` are then (next step) applied to the values over time.

Reply:
if you want the worst worker, but ignoring the worst 20% of the time, then you'd have

```
query="worker_max_tick|max",
over_time="quantile(.8)",
```